src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvokerProvider.cs

```diff
@@ -264,18 +264,44 @@ internal static void PopulateHandlerMethodDescriptors(TypeInfo type, CompiledPag
 
                 var formAction = new StringSegment(method.Name, formActionStart, formActionLength);
 
+                var methodParameters = method.GetParameters();
+                var parameters = new HandlerParameter[methodParameters.Length];
+
+                for (var j = 0; j < methodParameters.Length; j++)
+                {
+                    var parameter = methodParameters[j];
+
+                    parameters[j] = new HandlerParameter(parameter.Name, parameter.ParameterType, GetDefaultValue(parameter));
+                }
+
                 var handlerMethodDescriptor = new HandlerMethodDescriptor
                 {
                     Method = method,
                     Executor = ExecutorFactory.CreateExecutor(actionDescriptor, method),
                     FormAction = formAction,
                     HttpMethod = httpMethod,
+                    Parameters = parameters
                 };
 
                 actionDescriptor.HandlerMethods.Add(handlerMethodDescriptor);
             }
         }
 
+        private static object GetDefaultValue(ParameterInfo methodParameter)
```

Comment:
The fact that I've got this logic in two places makes me wonder if I've gotten an abstraction wrong.